backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -78,74 +83,160 @@ def clean_up_share(self, *kwargs):
     @staticmethod
     def build_access_point_name(share):
         S3AccessPointName = utils.slugify(
-            share.datasetUri + '-' + share.principalId,
-            max_length=50, lowercase=True, regex_pattern='[^a-zA-Z0-9-]', separator='-'
+            share.datasetUri + "-" + share.principalId,
+            max_length=50,
+            lowercase=True,
+            regex_pattern="[^a-zA-Z0-9-]",
+            separator="-",
         )
         logger.info(f"S3AccessPointName={S3AccessPointName}")
         return S3AccessPointName
 
+    def check_bucket_policy(self) -> None:
+        """
+        This function will check if delegate access to access point statemnt is in the bucket policy
+        and add to folder errors if check fails
+        :return: None
+        """
+        s3_client = S3Client(self.source_account_id, self.source_environment.region)
+        bucket_policy = s3_client.get_bucket_policy(self.bucket_name)
+        error = False
+        if not bucket_policy:
+            error = True
+        else:
+            bucket_policy = json.loads(bucket_policy)
+            counter = count()
+            statements = {item.get("Sid", next(counter)): item for item in bucket_policy.get("Statement", {})}
+
+            if DATAALL_DELEGATE_TO_ACCESS_POINT not in statements.keys():
+                error = True
+        if error:
```

Comment:
Minor: instead of returning a single error, we could append a more specific error for each case. If not bucket_policy --> something about the policy does not exists, in case it exists, ---> something about statement missing

Reply:
We have that logic already in `format_error_message()` - I can work to split it out into 2 different errors

Trying to keep error formatters as general as possible so we do not run down the path of trying to map each check / error with a new custom message